types/list.go

```diff
@@ -15,10 +15,28 @@ import (
 )
 
 var (
-	_ attr.Type  = ListType{}
-	_ attr.Value = &List{}
+	_ ListTypable  = ListType{}
+	_ ListValuable = &List{}
 )
 
+// ListTypable extends attr.Type for list type types.
+// Implement this interface to create a custom ListType type type.
+type ListTypable interface {
```

Comment:
Can you add changelog entries for these? e.g.

```
types: Added `ListTypable` and `ListValuable` interface types, which enable embedding existing list types for custom types
```

Or something along those lines?